petab/core.py

```diff
@@ -507,10 +512,90 @@ def _apply_overrides(
                 overrides, condition_ix,
                 row.observableId, override_type='noise')
 
+    fill_in_nominal_values(mapping, parameter_df)
+
     handle_missing_overrides(mapping, warn=warn_unmapped)
     return mapping
 
 
+def _apply_dynamic_parameter_overrides(mapping,
+                                       condition_df: pd.DataFrame,
+                                       parameter_df: pd.DataFrame,
+                                       par_sim_id_to_ix):
+    """Apply dynamic parameter overrides from condition table (inplace).
+
+    Arguments:
+        mapping, par_sim_id_to_ix:
+            see get_optimization_to_simulation_parameter_mapping
+        condition_df, parameter_df: PEtab condition and parameter table
+    """
+    for overridee_id in condition_df.columns:
+        if overridee_id == 'conditionName':
+            continue
+        if condition_df[overridee_id].dtype != 'O':
+            continue
+
+        for condition_idx, overrider_id \
+                in enumerate(condition_df[overridee_id]):
+            if isinstance(overridee_id, str):
+                _check_dynamic_parameter_override(
+                    overridee_id, overrider_id, parameter_df)
+                mapping[condition_idx][par_sim_id_to_ix[overridee_id]] = \
+                    overrider_id
+
+
+def _check_dynamic_parameter_override(
+        overridee_id, overrider_id, parameter_df: pd.DataFrame):
+    """Check for valid replacement of parameter overridee_id by overrider_id.
+    Matching scales, etc."""
+
+    if 'parameterScale' not in parameter_df:
+        return  # Nothing to check
+
+    # in case both parameter are in parameter table, their scale
+    # must match.
+    if overridee_id in parameter_df.index \
```

Comment:
Is the overridee_id not required to be in parameter_df? Where is that parameter-condition override documented?

Reply:
> Is the overridee_id not required to be in parameter_df? 

Wasn't completely sure how to handle that. If a parameter is always overridden: does is still have to be in the parameter table? We don't require it for the output parameters, so I think we shouldn't require it here. 

> Where is that parameter-condition override documented?

The documentation has a "parameterId" in the sample table in "Condition table" section. Could be extended at some point.

Reply:
Yes, I would then also not require the overridee id to be in the parameter table, since none of the columns makes sense really (boundaries, scale, ... all overridden)

Reply:
I would leave that in for now. This way things should be safe. To allow for non-matching parameter scale overrides, `get_optimization_to_simulation_scale_mapping ` will also have to be adapted, I think.